controllers/openstackmachine_controller.go

```diff
@@ -328,9 +337,24 @@ func (r *OpenStackMachineReconciler) reconcileNormal(ctx context.Context, scope
 		return ctrl.Result{}, err
 	}
 
-	instanceStatus, err := r.getOrCreate(scope.Logger(), cluster, openStackCluster, machine, openStackMachine, computeService, userData)
+	managedSecurityGroups := getManagedSecurityGroups(openStackCluster, machine, openStackMachine)
+	instanceTags := getInstanceTags(openStackMachine, openStackCluster)
+	portsStatus, err := r.getOrCreatePorts(scope.Logger(), clusterName, openStackCluster, openStackMachine.Spec.Ports, openStackMachine.Spec.Trunk, managedSecurityGroups, instanceTags, openStackMachine.Name, openStackMachine, networkingService)
 	if err != nil {
-		// Conditions set in getOrCreate
+		// Conditions set in getOrCreatePorts
+		return ctrl.Result{}, err
+	}
+	// TODO(emilien) do we want to deal with Ports Status like we do for the nova instance? Might be expensive...
+	// In the meantime, we consider that ports are ready if they are created.
+	conditions.MarkTrue(openStackMachine, infrav1.PortsReadyCondition)
+	portIDs := make([]string, len(portsStatus))
+	for i, port := range portsStatus {
+		portIDs[i] = port.ID
+	}
```

Comment:
Looks like we're doing this again, this could be a utility function.

Reply:
where?

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1788/files#diff-39b99bc838c9f76967790a10572cc44f0c41ccacddcd6e1eff934828d4b050a3R411-R413

Reply:
This indeed could be an utility function

https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1788/files#diff-39b99bc838c9f76967790a10572cc44f0c41ccacddcd6e1eff934828d4b050a3R428-R430